paymentsheet/src/main/java/com/stripe/android/paymentsheet/viewmodels/BaseSheetViewModel.kt

```diff
@@ -657,18 +658,24 @@ internal abstract class BaseSheetViewModel(
     }
 
     fun supportedPaymentMethodForCode(code: String): SupportedPaymentMethod {
-        val currentSelection = newPaymentSelection?.takeIf { it.paymentMethodCreateParams.typeCode == code }
-
         return requireNotNull(
             paymentMethodMetadata.value?.supportedPaymentMethodForCode(
                 code = code,
-                context = getApplication(),
-                paymentMethodCreateParams = currentSelection?.paymentMethodCreateParams,
-                paymentMethodExtraParams = currentSelection?.paymentMethodExtraParams,
             )
         )
     }
 
+    fun formElementsForCode(code: String): List<FormElement> {
+        val currentSelection = newPaymentSelection?.takeIf { it.paymentMethodCreateParams.typeCode == code }
+
+        return paymentMethodMetadata.value?.formElementsForCode(
+            code = code,
+            context = getApplication(),
+            paymentMethodCreateParams = currentSelection?.paymentMethodCreateParams,
+            paymentMethodExtraParams = currentSelection?.paymentMethodExtraParams,
+        ) ?: emptyList()
+    }
```

Comment:
If we only plan to call this from a UI context, what do you think about having the caller pass the context rather than use the application context? 

Might make it easier in the future for migrating `BaseSheetViewModel` off of `AndroidViewModel`

Reply:
Makes sense to me. Mind if we make that change when we remove the dependence on `AndroidViewModel`?